Jenkinsfile

```diff
@@ -11,39 +12,195 @@ def branchOrPR(pr) {
+
+    def returnMap = [:]
+
+    def benchmarks = (body =~ /\/benchmarks\/(\w+)\/(.*?)', (.*?), (.*?), (.*?), (.*?)\)/)
+    def compilation = (body =~ /compilation', (.*?), (.*?), (.*?), (.*?)\)/)[0]
+    def mean = (body =~ /Mean: (\d{1}\.?\d{11})/)[0][1]
+    def hash = (body =~ /Merge (.*?) into/)[0][1]
+    def current_os = (body =~ /Current OS: (.*?) !/)[0][1]
+
+    def cpu = ""
+    def gpp = ""
+    def clang = ""
+    def sys_ver = ""
+
+    if(current_os == "windows"){
+        cpu = (body =~ /(?s)wmic CPU get NAME(.*?)(C:|J:|Z:)/)[0][1]
+        sys_ver = (body =~ /(?s)>ver(.*?)(C:|J:|Z:)/)[0][1]
+        gpp = (body =~ /(?s)g\+\+ --version(.*?)(C:|J:|Z:)/)[0][1]
+        clang = (body =~ /(?s)clang --version(.*?)(C:|J:|Z:)/)[0][1]
     }
-    def result_match = (performance_log =~ /(?s)\).(\d{1}\.?\d{11})/)
-    try{
-        comment += "Result: " + result_match[0][1].toString() + "\\r\\n"
+    else if(current_os == "macos"){
+        cpu = (body =~ /(?s)sysctl -n machdep\.cpu\.brand_string(.*?)\+ sw_vers/)[0][1]
+        sys_ver = (body =~ /(?s)sw_vers(.*?)\+ g\+\+/)[0][1]
+        gpp = (body =~ /(?s)g\+\+ --version(.*?)\+ clang/)[0][1]
+        clang = (body =~ /(?s)clang --version(.*?)\+ echo/)[0][1]
     }
-    catch(Exception ex){
-        comment += "Result: " + "Regex did not match anything" + "\\r\\n"
+    else{
+        cpu = (body =~ /(?s)lscpu(.*?)\+ lsb_release/)[0][1]
+        sys_ver = (body =~ /(?s)lsb_release -a(.*?)\+ g\+\+/)[0][1]   
+        gpp = (body =~ /(?s)g\+\+ --version(.*?)\+ clang/)[0][1]
+        clang = (body =~ /(?s)clang --version(.*?)\+ echo/)[0][1]
     }
-    def result_match_hash = (performance_log =~ /Merge (.*?) into/)
-    try{
-        comment += "Commit hash: " + result_match_hash[0][1].toString() + "\\r\\n"
+
+    for (i = 0; i < benchmarks.size(); i++) {
+
+      def name = benchmarks[i][1]
+      def filename = benchmarks[i][2]
+      def old_value = benchmarks[i][3]
+      def new_value = benchmarks[i][4]
+      def ratio = benchmarks[i][5]
+      def change = benchmarks[i][6]
+
+      returnMap["$name"] = [
+        "old": old_value,
+        "new": new_value,
+        "ratio": ratio,
+        "change": change + (change.toDouble() > 0 ? "% faster" : "% slower")
+      ]
+
     }
-    catch(Exception ex){
-        comment += "Commit hash: " + "Regex did not match anything" + "\\r\\n"
+
+    returnMap["compilation"] = [
+        "old": compilation[1],
+        "new": compilation[2],
+        "ratio": compilation[3],
+        "change": compilation[4] + (compilation[4].toDouble() > 0 ? "% faster" : "% slower")
+      ]
+
+    returnMap["mean"] = mean.toString()
+    returnMap["hash"] = hash
+
+    returnMap["system"] = [
+        "cpu": escapeStringForJson(cpu),
+        "sys_ver": escapeStringForJson(sys_ver),
+        "gpp": escapeStringForJson(gpp),
+        "clang": escapeStringForJson(clang)
+      ]
+
+    return returnMap
+}
+
+def MapLastGitHubComment(body){
+    returnMap = [:]
```

Comment:
Ah, I think there was a big misunderstanding here. We don't want to do anything with previous GitHub comments. We just always want to post the relative comparisons that come out of running the cmdstan performance tests against Stan develop.

Reply:
Want to hop on the phone in a half hour or so? Just want to make sure we get the specs for this cleared up.

Reply:
[get_last_results](https://github.com/stan-dev/performance-tests-cmdstan/blob/a0d318f4970dc8605b2f3c728547321469864c3d/Jenkinsfile#L125) and [mapLastGitHubComment](https://github.com/stan-dev/performance-tests-cmdstan/blob/a0d318f4970dc8605b2f3c728547321469864c3d/Jenkinsfile#L106) are two functions I wrote and used a while ago, they're not used in the code anymore but left them there for possible future needs.
I can just remove them from this PR, won't affect anything.

Yes, I'm available anytime for the next 4~6 hours.